### PR TITLE
Add concurrency safety to store + various fixes

### PR DESCRIPTION
**Pull request by snario**
- Add a private `withLock` function to the store. Functions which call `this.backend.setChannel` should be wrapped with a `withLock` call to guarantee safety
- The `simple-hub` logs the exact string that it sends, as to not mislead developers when debugging
- The `simple-hub` serializes guarantee outcomes
- The `web3torrent-integration-test-with-hub` sets the same firebase prefix between `web3torrent` and `simple-hub`

TODO: 
- [ ] Devise a scheme to prevent calling `this.backend.setChannel` without first acquiring a log.
- [ ] Clean up the environment variable mess that we've created.
- [ ] Add explicit tests for multiple concurrent writes to channel entries?

### CODE REVIEW

**Comment by andrewgordstewart** [comment]
It would be nice to merge this as is, the benefits being:
1. we can hopefully sustain ✅ 's on `master`
2. we can resolve some possible unpredictable behaviour

**Comment by andrewgordstewart** [comment]
> Devise a scheme to prevent calling this.backend.setChannel without first acquiring a log.

Locally, I was playing around with changing the type of `this.backend.setChannel` to be
`(key: string, data: () => Promise<ChannelStoredData>)`.

Then, the backend is responsible for concurrency-safety.

File: packages/simple-hub/src/wallet/xstate-wallet-internals/index.ts [hunk 1]
```diff
@@ -171,18 +173,24 @@ function deserializeAllocationItem(allocationItem: AllocationItemWire): Allocati
   return {destination, amount: bigNumberify(amount)};
 }
 
-function serializeOutcome(outcome: Outcome): OutcomeWire {
+export function serializeOutcome(outcome: Outcome): OutcomeWire {
```
Comment (kerzhner) [inline packages/simple-hub/src/wallet/xstate-wallet-internals/index.ts:176]: Does this need to be exported?

File: packages/simple-hub/src/wallet/xstate-wallet-internals/index.ts [hunk 2]
```diff
@@ -309,3 +317,13 @@ export const firstState = (
   participants,
   outcome
 });
+
+export function makeDestination(addressOrDestination: string): string {
```
Comment (kerzhner) [inline packages/simple-hub/src/wallet/xstate-wallet-internals/index.ts:321]: Should [`deposit.ts`](https://github.com/statechannels/monorepo/blob/ags/wallet-fixes/packages/simple-hub/src/wallet/deposit.ts#L35) be updated to use this function?

**Comment by snario** [comment]
@andrewgordstewart It looks like the test failed, but the job succeeded

https://app.circleci.com/pipelines/github/statechannels/monorepo/4987/workflows/d19ec482-f2b4-443d-8e11-262db7e86dd6/jobs/18094

```
Killed
error Command failed with exit code 137.
```

**Comment by andrewgordstewart** [comment]
Xstate wallet did start. Did you read the logs in the puppeteer test?

**Pull request merged**